src/main/java/io/debezium/connector/jdbc/dialect/mysql/MySqlDatabaseDialect.java

```diff
@@ -141,7 +141,7 @@ public String getFormattedTimestamp(TemporalAccessor value) {
     @Override
     public String getFormattedTimestampWithTimeZone(String value) {
         final ZonedDateTime zonedDateTime = ZonedDateTime.parse(value, ZonedTimestamp.FORMATTER);
```

Comment:
MySQL save Timestamp as UTC, so if we format time for field's default, we shoule convert it to UTC?

Reply:
MySQL stores `timestamps` as `UTC` but in retrieving it converts to the current timezone.
https://dev.mysql.com/doc/refman/8.0/en/datetime.html

Reply:
> MySQL stores `timestamps` as `UTC` but in retrieving it converts to the current timezone. https://dev.mysql.com/doc/refman/8.0/en/datetime.html

Yes, however this function used for DDL.
When create table with default timestamp, i think it should be converted to utc.

Reply:
I'm afraid this does not solve the issue, see the following on `mysql:latest`:
```
mysql> create table test1 (data timestamp(6) default '1970-01-01T00:00:00' not null);
ERROR 1067 (42000): Invalid default value for 'data'
mysql> create table test1 (data timestamp(6) default '1970-01-01 00:00:00' not null);
ERROR 1067 (42000): Invalid default value for 'data'
```

The issue here is rather one with MySQL configuration.  When you specify unix epoch, `1970-01-01 00:00:00`, MySQL translates this to `0` seconds since unix epoch and MySQL explicitly reserves 0 seconds since epoch to represent the value of `0000-00-00 00:00:00` or "zero" timestamp.

I'll update the Jira issue with this information, FWIW, if you specify a default value with at least 1 second since unix epoch such as `1970-01-01 00:00:01`, MySQL works just fine, see below with both formats:

```
mysql> create table test1 (data timestamp(6) default '1970-01-01T00:00:01' not null);
Query OK, 0 rows affected (0.03 sec)

mysql> create table test2 (data timestamp(6) default '1970-01-01 00:00:01' not null);
Query OK, 0 rows affected (0.02 sec)
```

So there is no need to change the format here, it's the actual epoch value and how MySQL interprets it.

Reply:
@Naros 
Yes, i agree 00:00:01 is better than 00:00:00.

However, i think "1970-01-01 00:00:01" is better than "1970-01-01T00:00:01".

Although [The date and time parts can be separated by T rather than a space. For example, '2012-12-31 11:30:45' '2012-12-31T11:30:45' are equivalent.](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-literals.html)

However, most people and the official documentation would argue that T should not be used.
MySQL recognizes [DATETIME](https://dev.mysql.com/doc/refman/8.0/en/datetime.html) and [TIMESTAMP](https://dev.mysql.com/doc/refman/8.0/en/datetime.html) values in these formats: As a string in either 'YYYY-MM-DD hh:mm:ss' or 'YY-MM-DD hh:mm:ss' format. 

Other discovery
- session's time_zone will affect server understand default
```sql
mysql> SELECT @@session.time_zone;
+---------------------+
| @@session.time_zone |
+---------------------+
| Asia/Shanghai       |
+---------------------+
1 row in set (0.02 sec)

mysql> create table test3 (data timestamp(6) default '1970-01-01 00:00:01' not null);
ERROR 1067 (42000): Invalid default value for 'data'
mysql> create table test3 (data timestamp(6) default '1970-01-01 08:00:01' not null);
Query OK, 0 rows affected (0.02 sec)
```